test/baselines.js

```diff
@@ -23,29 +25,47 @@ function build(file) {
 
 describe('baselines', () => {
   files.forEach(file => {
-    const local = LOCAL_DIR + file;
     const reference = REFERENCE_DIR + file;
-    file = SOURCES_DIR + file;
-    it(file, async () => {
-      let spec = await build(file);
-      const contents = spec.toHTML();
-      let str;
-      try {
-        str = fs.readFileSync(reference, 'utf8');
-      } catch (e) {
-        // ignored
+    it(SOURCES_DIR + file, async () => {
+      let spec = await build(SOURCES_DIR + file);
+
+      let expectedFiles = new Map();
+
+      (function walk(f) {
+        if (!fs.existsSync(f)) {
+          return;
+        }
+        if (fs.lstatSync(f).isDirectory()) {
+          for (let file of fs.readdirSync(f)) {
+            walk(path.join(f, file));
+          }
+        } else {
+          expectedFiles.set(path.relative(reference, f), fs.readFileSync(f, 'utf8'));
+        }
+      })(reference);
+
+      let actualFiles = spec.generatedFiles;
+      if (actualFiles.size === 1 && actualFiles.has(null)) {
+        // i.e. single-file output
+        actualFiles.set('', actualFiles.get(null));
+        actualFiles.delete(null);
       }
-      if (contents !== str) {
-        try {
-          fs.mkdirSync(LOCAL_DIR);
-        } catch (e) {
-          // ignored
+
+      let threw = true;
+      try {
+        if (expectedFiles.size === 0) {
+          throw new Error('Baseline does not exist');
+        }
+        assert.deepStrictEqual(actualFiles, expectedFiles);
+        threw = false;
+      } finally {
+        if (threw || true) {
```

Comment:
?

Reply:
Whoops, left over from testing it. Fixed.